src/lib/cpu-clk-manager.c

```diff
@@ -62,14 +61,9 @@ int core_kcps_adjust(int adjusted_core_id, int kcps_delta)
 	/* set clock according to maximum requested mcps budget */
 	freq = max_core_consumption();
 
-	for (core_id = 0; core_id < CONFIG_CORE_COUNT; core_id++) {
-		/* Convert kcps to cps */
-		ret = request_freq_change(core_id, freq * 1000);
-		if (ret < 0)
-			goto out;
-	}
+	/* Convert kcps to cps */
+	ret = request_freq_change(adjusted_core_id, freq * 1000);
```

Comment:
That is correct in the sense, that there is only one register setting clocks for all of the cores (for cavs and ace at least), but please double check if this call updates all of the `clock_info` structures. `clock_set_freq` does not explicitly do that. For sofs CAVS implementation it works fine, as the platform specific `clk_info->set_freq` function iterates over cores. When using zephyrs implementation though, there is a separate structure holding clock info (see zephyr/soc/xtensa/intel_adsp/common/clk.c). This used to be unaligned with sofs structures and might be worth checking.

Reply:
Good catch, thank you @KFrydryk !
Indeed, Zephyr stores clock info per Core:
`static struct adsp_clock_info platform_clocks[CONFIG_MP_MAX_NUM_CPUS];`

It does not make too much sense and it is confusing. Also potentially may lead to some issues with clock management.
I think this should be investigated separately then, with another PR's for SOF and Zephyr.

I am going to remove this commit from this PR, as it is not critical for the moment.